modules/API/pages/built-in-endpoints.adoc

```diff
@@ -1297,9 +1297,10 @@ This endpoint upserts vertices and/or edges into a graph. To upsert means that i
 
 ==== Atomic upsert transaction
 
-By default, the `+POST /graph/{graph_name}+` endpoint is not atomic. If something goes wrong during the process of the request,  the request data can be partially consumed by the database.
+By default, the `+POST /graph/{graph_name}+` endpoint is not atomic. If something goes wrong during the process of the request, the request data can be partially consumed by the database.
 
-You can append a query string parameter `atomic_post` to the URL of the request and set its value to true to make the request an atomic transaction, which means that updates to the database contained in the request are all-or-nothing. Either all changes are successful, or none is successful.
+You can append a request header `gsql-atomic-level` to the request and set its value to `atomic` to make the request an atomic transaction.
```

Comment:
Can you also add that the user have the option to specify that a request is non-atomic by setting header ```gsql-atomic-level``` to ```nonatomic```?

Reply:
@aozeng I was under the impression that `nonatomic` is the default behavior. Do you just want to share this as an option? Or does this apply to some other atomic endpoints and can make them not atomic?

Reply:
Yes, I'd like to list the option to use `nonatomic` for `upsert`, even though it does not change the default behavior.

The user might want to explicitly use this `nonatomic` value in their REST request to stress that the choice of the  non atomic `upsert`  is intended and to inform the code viewer the existence of a header for atomic `upsert`.